src/java.base/share/classes/java/lang/annotation/ElementType.java

```diff
@@ -26,106 +26,213 @@
+     * declaration statement, or declared within the header of a {@code
+     * for} or {@code try} statement, or within a pattern as a pattern
+     * variable. However, the similar case of an exception variable
+     * declared in a {@code catch} clause is considered a {@link
+     * #PARAMETER} instead.
+     */
     LOCAL_VARIABLE,
 
-    /** Annotation interface declaration (Formerly known as an annotation type.) */
+    /** 
+     * The declaration of an annotation interface (a subcategory of
+     * {@link #TYPE}). An annotation that itself appears on the
+     * declaration of an annotation interface is sometimes informally
+     * called a "meta-annotation"; {@link Target} itself is a primary
+     * example.
+     */
     ANNOTATION_TYPE,
 
-    /** Package declaration */
+    /**
+     * The declaration of a package in a {@code package-info.java} file.
+     * Package declarations in other source files cannot be annotated.
+     *
+     * @jls 7.4 Package Declarations
+     */
     PACKAGE,
 
     /**
-     * Type parameter declaration
+     * The declaration of a type parameter within a generic class,
+     * method, or constructor declaration.
      *
      * @since 1.8
      */
     TYPE_PARAMETER,
 
     /**
-     * Use of a type
+     * A code location where a compile-time type is being indicated.
+     * An annotation in such a location is called a "type-use
+     * annotation" (sometimes called just "type annotation", but be
+     * careful not to confuse this with {@link #TYPE}, which is only
+     * for <em>class</em> declarations).
+     * 
+     * <p>This is a very broad category: {@jls 4.11} lists seventeen
+     * kinds of type contexts, followed by five more locations where
+     * type-use annotations can also legally appear. Six of these
+     * locations are also annotatable <em>declarations</em>
+     * themselves; see <a href="#ambiguous">ambiguous cases</a> above.
+     * 
+     * <p>In addition, specifying this target automatically includes
+     * the declaration targets {@link #TYPE} and {@link
+     * #TYPE_PARAMETER}, though these are not type contexts.
      *
      * @since 1.8
      */
     TYPE_USE,
 
     /**
-     * Module declaration.
+     * The declaration of a module in a {@code module-info.java} file.
      *
      * @since 9
+     * @jls 7.7 Module Declarations
      */
     MODULE,
 
     /**
-     * Record component
+     * The declaration of a record component, in the header of a record
+     * class declaration. If an annotation interface should apply
+     * conceptually to the field, accessor method, or constructor
+     * parameter that is generated corresponding to a record component,
+     * it should specify the appropriate element types ({@link #FIELD},
+     * {@link #METHOD}, or {@link #PARAMETER}), instead of {@code
+     * RECORD_COMPONENT} or in addition to it. This allows the
+     * annotation to be automatically copied to any generated elements
+     * it applies to.
```

Comment:
Doesn't this effectively become another "ambiguous context?" Note that in both cases, the observation of the annotations via the `class` file format (and subsequently, the core reflection API) is inconsistent.

Reply:
Well, this is one of the ambiguous contexts listed above. Not sure if I'm reading this right. What's the "inconsistency" you mean?

Reply:
Say you have a record component like:
`record Sample(@Anno int val)`
Where `@Anno` applies to `RECORD_COMPONENT` and `FIELD`.

Then we can observe this annotation on the record component or its field, but not its getter method. This creates a scenario similar to how people can observe an annotation on both the type-use and the declaration for some common annotations like `@Nullable`, yet some annotations are only present on one of the sites.

Reply:
Okay, I see the connection. I'm trying to think about why I used the word "ambiguous" to talk about the case above. I think there are 2 reasons

1. For purposes of answering "which of the two kinds of annotations is this, anyway?" But in this case, it's not so much that it's ambiguous, it's just that its *both*.
2. Specifically because of the `@DualPurposeAnnot String[] field;` problem. It's maybe "ambiguous" whether this is trying to say something about the field or the _component_ type of the array type.

I think I'll try describing this situation without "ambiguous".

Getting back to record/field, I don't see a problem as long as any annotation interface that targets both kinds doesn't try to document incompatible *meanings* for the two.